### ISSUE TITLE
Does yolov8 associate bounding box sizes and shapes with the classes he is learning in training

### ISSUE THREAD

**Issue by IGlace** [ultralytics/ultralytics#22876]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I'm training a yolov8 object detection model on a dataset of 4 classes, all classes are for humans, i'm just trying to make the model learn the discriminitive features between those humans to classify properly those 4 classes, among those discriminitive features some classes, are know for their unique and repetitive bounding box shape and size, and other classes contain a lot of different bounding boxes shapes.
Now my question is does Yolov8 model, look at the bounding box features overall to classify it only or are the shape and size of bounding boxes included in those features that he learn ?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @IGlace, thank you for your interest in Ultralytics 🚀! This is an **automated assistant response** to help you get started while an Ultralytics engineer reviews your question and follows up soon 🤝

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered, including details on model behavior, training, and customization.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (like your YOLOv8 human-class discrimination and bounding box–related query), please provide as much information as possible, including:
- Dataset image examples or structure description
- Brief details about your annotations and classes
- Training command, key hyperparameters, and logs
- Any observations or artifacts from predictions that illustrate your concern

Also verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/), as these often resolve many training and behavior-related questions.

Join the Ultralytics community where it suits you best:
- For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧
- For in-depth discussions, check out [Discourse](https://community.ultralytics.com/)
- Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community

## Upgrade

Before proceeding, please ensure you are on the latest version of `ultralytics`, as behavior and features may change between releases. Upgrade the `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a **Python>=3.8** environment with **PyTorch>=1.8** to verify your issue or question is not related to an outdated version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
It can learn it if there's consistency. But it's unlikely to occur because the augmentations change zoom levels and the size of the boxes also would change depending on camera angles

**Comment by IGlace** [comment]
> It can learn it if there's consistency. But it's unlikely to occur because the augmentations change zoom levels and the size of the boxes also would change depending on camera angles

But i normally don't use data augmentation at all in my training, even if i do i usually keep it very minimal, also most of the data in my dataset is cctv data that is normally in the corner of the room or generally somewhere high above ground level, but also not too high like a fish eye camera from above.
Does your explanation still apply in this case ?

**Comment by Y-T-G** [comment]
Ultralytics has augmentations enabled by default including mosaic and scale that changes zoom levels. But if you disable them and the size is consistent across all your labels, the model can learn the classes based on bounding box size

**Comment by IGlace** [comment]
> Ultralytics has augmentations enabled by default including mosaic and scale that changes zoom levels. But if you disable them and the size is consistent across all your labels, the model can learn the classes based on bounding box size

Ok thank you very much for your time that help a lot.

**Comment by glenn-jocher** [comment]
Glad it helped! If you later want to control how much the model can rely on box size vs. appearance, you can tune or disable augmentations like `mosaic`, `scale`, `mixup`, etc. in your training config; the available options and defaults are listed in the [augmentation configuration docs](https://docs.ultralytics.com/usage/cfg/#augmentation-settings).
